File: android/src/main/java/com/glia/widgets/ionic/GliaSdkPlugin.java
package com.glia.widgets.ionic;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "GliaSdk")
public class GliaSdkPlugin extends Plugin {

    private final GliaSdk implementation = new GliaSdk(this::notifyListeners);

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }

    @PluginMethod
    public void configure(PluginCall call) {
        implementation.configure(call, getActivity());
    }

    @Deprecated
    @PluginMethod
    public void presentEntryWidget(PluginCall call) {
        implementation.presentEntryWidget(call, getActivity());
    }

    @PluginMethod
    public void showEntryWidget(PluginCall call) {
        implementation.showEntryWidget(call, getActivity());
    }

    @PluginMethod
    public void hideEntryWidget(PluginCall call) {
        implementation.hideEntryWidget(call);
    }

    @PluginMethod
    public void startChat(PluginCall call) {
        implementation.startChat(call, getActivity());
    }

    @PluginMethod
    public void startAudio(PluginCall call) {
        implementation.startAudio(call, getActivity());
    }

    @PluginMethod
    public void startVideo(PluginCall call) {
        implementation.startVideo(call, getActivity());
    }

    @Deprecated
    @PluginMethod
    public void startSecureConversation(PluginCall call) {
        implementation.startSecureConversation(call, getActivity());
    }

    @PluginMethod
    public void startSecureMessaging(PluginCall call) {
        implementation.startSecureMessaging(call, getActivity());
    }

    @PluginMethod
    public void clearVisitorSession(PluginCall call) {
        implementation.clearVisitorSession(call);
    }

    @Deprecated
    @PluginMethod
    public void listQueues(PluginCall call) {
        implementation.listQueues(call);
    }

    @PluginMethod
    public void getQueues(PluginCall call) {
        implementation.getQueues(call);
    }

    @Deprecated
    @PluginMethod
    public void showVisitorCodeViewController(PluginCall call) {
        implementation.showVisitorCode(call);
    }

    @PluginMethod
    public void showVisitorCode(PluginCall call) {
        implementation.showVisitorCode(call);
    }

    @PluginMethod
    public void hideVisitorCode(PluginCall call) {
        implementation.hideVisitorCode(call);
    }

    @PluginMethod
    public void authenticate(PluginCall call) {
        implementation.authenticate(call);
    }

    @PluginMethod
    public void deauthenticate(PluginCall call) {
        implementation.deauthenticate(call);
    }

    @PluginMethod
    public void isAuthenticatedInternal(PluginCall call) {
        implementation.isAuthenticated(call);
    }

    @PluginMethod
    public void refreshAuthentication(PluginCall call) {
        implementation.refreshAuthentication(call);
    }

    @PluginMethod
    public void pauseLiveObservation(PluginCall call) {
        implementation.pauseLiveObservation(call);
    }

    @PluginMethod
    public void resumeLiveObservation(PluginCall call) {
        implementation.resumeLiveObservation(call);
    }

    @PluginMethod
    public void getVisitorInfo(PluginCall call) {
        implementation.getVisitorInfo(call);
    }

    @PluginMethod
    public void updateVisitorInfo(PluginCall call) {
        implementation.updateVisitorInfo(call);
    }

    @PluginMethod
    public void endEngagement(PluginCall call) {
        implementation.endEngagement(call);
    }

    @PluginMethod
    public void subscribeToPushNotificationTypes(PluginCall call) {
        implementation.subscribeToPushNotificationTypes(call, getActivity());
    }
}

File: ios/Sources/GliaSdkPlugin/GliaSdkPlugin.swift
import Capacitor
import Foundation

/// Please read the Capacitor iOS Plugin Development Guide
/// here: https://capacitorjs.com/docs/plugins/ios
@objc(GliaSdkPlugin)
public class GliaSdkPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "GliaSdkPlugin"
    public let jsName = "GliaSdk"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "configure", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "presentEntryWidget", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "showEntryWidget", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "hideEntryWidget", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "startChat", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "startAudio", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "startVideo", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "clearVisitorSession", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "listQueues", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "getQueues", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "authenticate", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "deauthenticate", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "isAuthenticated", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "refreshAuthentication", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "showVisitorCodeViewController", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "showVisitorCode", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "startSecureConversation", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "startSecureMessaging", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "pauseLiveObservation", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "resumeLiveObservation", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(
            name: "subscribeToPushNotificationTypes",
            returnType: CAPPluginReturnPromise
        ),
        CAPPluginMethod(name: "getVisitorInfo", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "updateVisitorInfo", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = GliaSdk()

    @objc func configure(_ call: CAPPluginCall) {
        implementation.configure(call)
    }

    @objc func presentEntryWidget(_ call: CAPPluginCall) {
        implementation.showEntryWidget(call)
    }

    @objc func showEntryWidget(_ call: CAPPluginCall) {
        implementation.showEntryWidget(call)
    }

    @objc func hideEntryWidget(_ call: CAPPluginCall) {
        implementation.hideEntryWidget(call)
    }

    @objc func startChat(_ call: CAPPluginCall) {
        implementation.startChat(call)
    }

    @objc func startAudio(_ call: CAPPluginCall) {
        implementation.startAudio(call)
    }

    @objc func startVideo(_ call: CAPPluginCall) {
        implementation.startVideo(call)
    }

    @objc func clearVisitorSession(_ call: CAPPluginCall) {
        implementation.clearVisitorSession(call)
    }

    @objc func listQueues(_ call: CAPPluginCall) {
        implementation.getQueues(call)
    }

    @objc func getQueues(_ call: CAPPluginCall) {
        implementation.getQueues(call)
    }

    @objc func authenticate(_ call: CAPPluginCall) {
        implementation.authenticate(call)
    }

    @objc func deauthenticate(_ call: CAPPluginCall) {
        implementation.deauthenticate(call)
    }

    @objc func isAuthenticated(_ call: CAPPluginCall) {
        implementation.isAuthenticated(call)
    }

    @objc func refreshAuthentication(_ call: CAPPluginCall) {
        implementation.refreshAuthentication(call)
    }

    @objc func showVisitorCodeViewController(_ call: CAPPluginCall) {
        implementation.showVisitorCode(call)
    }

    @objc func showVisitorCode(_ call: CAPPluginCall) {
        implementation.showVisitorCode(call)
    }

    @objc func startSecureConversation(_ call: CAPPluginCall) {
        implementation.startSecureMessaging(call)
    }

    @objc func startSecureMessaging(_ call: CAPPluginCall) {
        implementation.startSecureMessaging(call)
    }

    @objc func pauseLiveObservation(_ call: CAPPluginCall) {
        implementation.pauseLiveObservation(call)
    }

    @objc func resumeLiveObservation(_ call: CAPPluginCall) {
        implementation.resumeLiveObservation(call)
    }

    @objc func subscribeToPushNotificationTypes(_ call: CAPPluginCall) {
        implementation.subscribeToPushNotificationTypes(call)
    }

    @objc func getVisitorInfo(_ call: CAPPluginCall) {
        implementation.getVisitorInfo(call)
    }

    @objc func updateVisitorInfo(_ call: CAPPluginCall) {
        implementation.updateVisitorInfo(call)
    }
}
